airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/context/env/ConnectorConfigurationPropertySource.kt

```diff
@@ -0,0 +1,147 @@
+        const val CONNECTOR_CONFIG_PREFIX: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.config"
+        const val CONNECTOR_CATALOG_PREFIX: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.catalog"
+        const val CONNECTOR_CATALOG_KEY: String = "json"
+        const val CONNECTOR_STATE_PREFIX: String = "$ROOT_CONFIGURATION_PROPERTY_KEY.state"
+        const val CONNECTOR_STATE_KEY: String = "json"
+
+        private fun resolveValues(commandLine: CommandLine): Map<String, Any> {
+            val values: MutableMap<String, Any> = mutableMapOf()
+            if (commandLine.rawArguments.isNotEmpty()) {
+                values[CONNECTOR_OPERATION] = commandLine.rawArguments[0].lowercase()
+                commandLine.optionValue(JavaBaseConstants.ARGS_CONFIG_KEY)?.let {
+                    values.putAll(
+                        loadFile(
+                            it as String,
+                            CONNECTOR_CONFIG_PREFIX,
+                        ),
+                    )
+                }
+                commandLine.optionValue(JavaBaseConstants.ARGS_CATALOG_KEY)?.let {
+                    values.putAll(
+                        loadFileContents(
+                            it as String,
+                            String.format(
+                                PREFIX_FORMAT,
+                                CONNECTOR_CATALOG_PREFIX,
+                                CONNECTOR_CATALOG_KEY
+                            ),
+                        ),
+                    )
+                }
+                commandLine.optionValue(JavaBaseConstants.ARGS_STATE_KEY)?.let {
+                    values.putAll(
+                        loadFileContents(
+                            it as String,
+                            String.format(
+                                PREFIX_FORMAT,
+                                CONNECTOR_STATE_PREFIX,
+                                CONNECTOR_STATE_KEY
+                            ),
+                        ),
+                    )
+                }
+            }
+            logger.debug { "Resolved values: $values" }
+            return values
+        }
+
+        private fun loadFile(
+            propertyFilePath: String,
+            prefix: String,
+        ): Map<String, Any> {
+            if (propertyFilePath.isNotBlank()) {
+                val propertyFile = Path.of(propertyFilePath).toFile()
+                if (propertyFile.exists()) {
+                    val properties: Map<String, Any> =
+                        Jsons.deserialize(propertyFile.readText(), MapTypeReference())
+                    return flatten(properties, prefix).map { it.key to it.value }.toList().toMap()
+                } else {
+                    logger.warn { "Property file '$propertyFile', not found for prefix '$prefix'." }
+                    return mapOf()
+                }
+            } else {
+                return mapOf()
```

Comment:
would it be unexpected that the property file path is blank? (should we at least log something here?)